### DISCUSSION TITLE
CouchDB performance in single-node vs 1 node cluster

### DISCUSSION THREAD

**Discussion by unlikelyzero** [apache/couchdb#4567] [Q&A]
We are looking at improving our recommended couchdb configuration for some of our userbase to increase performance.

https://github.com/nasa/openmct/tree/master/src/plugins/persistence/couch

We were wondering if there's a performance difference when configuring a couchdb cluster for single-node vs a 1 node cluster.

**Comment by nickva** [comment]
I don't think there would be much of a performance difference. The setup might do slightly different things, create some default databases and such https://docs.couchdb.org/en/stable/setup/single-node.html

But there are some performance tweaks you can make, such as pick a default sharding value (Q), for instance. It defaults to 2 but if you have more than 2 CPUs available and expect to have fewer but large database, could start with a larger values like 8 or 32, for instance. Then potentially with enough CPU resources, your views or compactions would go 4 or 16 times faster than the default.

**Reply by unlikelyzero** [reply]
Got it, thanks!
